screens/LoginSignUp.js

```diff
@@ -0,0 +1,236 @@
+import React, {useState, useRef, useEffect} from 'react';
+import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
+import {
+  StyleSheet,
+  Text,
+  View,
+  Image,
+  TouchableOpacity,
+  TouchableHighlight,
+  TextInput,
+  Linking,
+  Alert,
+} from 'react-native';
+import auth from '@react-native-firebase/auth';
+
+const LoginSignUp = props => {
+  const [isLogin, setVisible] = useState(true);
+  const inputPasswordRef = useRef(null);
+  const [user, setUser] = useState(null);
+  const [email, setEmail] = useState(null);
+  const [password, setPassword] = useState(null);
+
+  function isStateChanged(user) {
+    setUser(user);
+  }
+
+  useEffect(() => {
+    const subscriber = auth().onAuthStateChanged(isStateChanged);
```

Comment:
This is a duplicate, similar code is in `App.js`
Read on `useContext`. Using that, we can share state values across multiple react components.